File: jvm-runtime/testdata/java/javaserver/src/main/java/xyz/block/ftl/enums/Animal.java
package xyz.block.ftl.enums;

import com.fasterxml.jackson.annotation.JsonIgnore;

import xyz.block.ftl.Enum;

/**
 * Comment on TypeEnum
 */
@Enum
public interface Animal {
    @JsonIgnore
    boolean isCat();

    @JsonIgnore
    boolean isDog();

    @JsonIgnore
    Cat getCat();

    @JsonIgnore
    Dog getDog();
}

File: jvm-runtime/testdata/kotlin/kotlinserver/src/main/kotlin/xyz/block/ftl/test/Animal.kt
package xyz.block.ftl.test

import com.fasterxml.jackson.annotation.JsonIgnore
import xyz.block.ftl.Enum

@Enum
interface Animal {
  @JsonIgnore
  fun isCat(): Boolean

  @JsonIgnore
  fun isDog(): Boolean

  @JsonIgnore
  fun getCat(): Cat

  @JsonIgnore
  fun getDog(): Dog
}
